### PR TITLE
updated gradles, dependencies, plugins, and code

### PR DESCRIPTION
**Pull request by TacoTheDank**
- java 1.8 compatibility
- build tools to 28
- gradle to 4.10.1
- updated some code to go with java 1.8

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,6 +12,7 @@ android:
   components:
     - platform-tools
     - tools
+    - tools
```
Comment (xeffyr) [inline .travis.yml]: Why we need second line with `- tools` ?

Comment (TacoTheDank) [reply .travis.yml]: from what i can remember, it ensures that it uses the most up-to-date sdk

Comment (xeffyr) [reply .travis.yml]: @TacoTheDank Actually, Travis ignores the duplicated lines.

**Comment by fornwall** [comment]
I'm closing this as there are quite some conflicts now.

@TacoTheDank Some of the changes look good, but please try to split up unrelated changes into several PR:s if possible, that makes reviewing and integrating the changes a lot easier!

**Comment by TacoTheDank** [comment]
gotchu :)

**Pull request closed without merging**